src/tangent_types/abstract_zero.jl

```diff
@@ -30,12 +27,52 @@ Base.convert(::Type{T}, x::AbstractZero) where {T<:Number} = zero(T)
 (::Type{Complex})(x::AbstractZero, y::Real) = Complex(false, y)
 (::Type{Complex})(x::Real, y::AbstractZero) = Complex(x, false)
 
-Base.getindex(z::AbstractZero, k) = z
-
+Base.getindex(z::AbstractZero, ind...) = z
 Base.view(z::AbstractZero, ind...) = z
 Base.sum(z::AbstractZero; dims=:) = z
 Base.reshape(z::AbstractZero, size...) = z
 
+# LinearAlgebra
+for f in (
+        :adjoint, :transpose, :Adjoint, :Transpose, :Diagonal,
+        :UpperTriangular, :LowerTriangular, :UnitUpperTriangular, :UnitLowerTriangular,
+        :UpperHessenberg
+    )
+    @eval LinearAlgebra.$f(z::AbstractZero) = z
+end
+for f in (:Symmetric, :Hermitian)
+    @eval LinearAlgebra.$f(z::AbstractZero, uplo=:U) = z
+end
+
+LinearAlgebra.Bidiagonal(dv::AbstractVector, ev::AbstractZero, uplo::Symbol) = Diagonal(dv)
+function LinearAlgebra.Bidiagonal(dv::AbstractZero, ev::AbstractVector, uplo::Symbol)
+    dv = fill!(similar(ev, length(ev) + 1), 0) # can't avoid making a dummy array
+    Bidiagonal(dv, convert(typeof(dv), ev), uplo)
+end
+LinearAlgebra.Bidiagonal(dv::AbstractZero, ev::AbstractZero, uplo::Symbol) = NoTangent()
+
+# one Zero:
+LinearAlgebra.Tridiagonal(dl::AbstractZero, d::AbstractVector, du::AbstractZero) = Diagonal(d)
+LinearAlgebra.Tridiagonal(dl::AbstractZero, d::AbstractVector, du::AbstractVector) = Bidiagonal(d, du, :U)
+LinearAlgebra.Tridiagonal(dl::AbstractVector, d::AbstractVector, du::AbstractZero) = Bidiagonal(d, dl, :L)
+function LinearAlgebra.Tridiagonal(dl::AbstractVector, d::AbstractZero, du::AbstractVector)
+    d = fill!(similar(dl, length(dl) + 1), 0)
+    Tridiagonal(convert(typeof(d), dl), d, convert(typeof(d), du))
+end
+# two Zeros:
+LinearAlgebra.Tridiagonal(dl::AbstractZero, d::AbstractVector, du::AbstractZero) = Diagonal(d)
+LinearAlgebra.Tridiagonal(dl::AbstractZero, d::AbstractZero, du::AbstractVector) = Bidiagonal(d, du, :U)
```

Comment:
Here, accepting Diagonal and Bidiagonal as the gradients of Tridiagonal avoids many cases of making an empty array to pad out the type. These types only accept same-type vectors, so you cannot pad it with a `Fill`.